src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -67,106 +65,93 @@ public StudyService(
+    Flux<StudyInfos> getStudyList() {
+        Flux<Study> studyList = studyRepository.findAll();
+        return studyList.map(study -> new StudyInfos(study.getName(), study.getDescription(), study.getCaseFormat()));
     }
 
-    void createStudy(String studyName, UUID caseUuid, String description) {
-        NetworkInfos networkInfos = persistentStore(caseUuid);
-        String caseFormat = getCaseFormat(caseUuid);
-        Study study = new Study(studyName, networkInfos.getNetworkUuid(), networkInfos.getNetworkId(), description, caseFormat, caseUuid, false);
-        studyRepository.insert(study);
+    Mono<Study> createStudy(String studyName, UUID caseUuid, String description) {
+        Mono<NetworkInfos> networkInfos = persistentStore(caseUuid);
+        Mono<String> caseFormat = getCaseFormat(caseUuid);
+
+        return Mono.zip(networkInfos, caseFormat)
+                .flatMap(t -> {
+                    Study study = new Study(studyName, t.getT1().getNetworkUuid(), t.getT1().getNetworkId(), description, t.getT2(), caseUuid, false);
+                    return studyRepository.insert(study);
+                });
     }
 
-    private String getCaseFormat(UUID caseUuid) {
-        String path = UriComponentsBuilder.fromPath("/" + CASE_API_VERSION + "/cases/{caseUuid}/format")
+    private Mono<String> getCaseFormat(UUID caseUuid) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + CASE_API_VERSION + "/cases/{caseUuid}/format")
                 .buildAndExpand(caseUuid)
                 .toUriString();
 
-        ResponseEntity<String> responseEntity = caseServerRest.exchange(path,
-                HttpMethod.GET,
-                HttpEntity.EMPTY,
-                String.class);
-
-        return responseEntity.getBody();
+        return webClient.get()
+                .uri(caseServerBaseUri + path)
+                .retrieve()
+                .bodyToMono(String.class);
     }
 
-    void createStudy(String studyName, MultipartFile caseFile, String description) throws IOException {
-        UUID caseUUid = importCase(caseFile);
-        NetworkInfos networkInfos = persistentStore(caseUUid);
-        String caseFormat = getCaseFormat(caseUUid);
-        Study study = new Study(studyName, networkInfos.getNetworkUuid(), networkInfos.getNetworkId(), description, caseFormat, caseUUid, true);
-        studyRepository.insert(study);
+    Mono<Study> createStudy(String studyName, MultipartFile caseFile, String description) {
+        Mono<UUID> caseUUid;
+        try {
+            caseUUid = importCase(caseFile);
+        } catch (IOException e) {
+            return Mono.error(new StudyException("error when importing the case"));
+        }
+        return caseUUid.flatMap(uuid -> {
+            Mono<NetworkInfos> networkInfos = persistentStore(uuid);
+            Mono<String> caseFormat = getCaseFormat(uuid);
+            return Mono.zip(networkInfos, caseFormat)
+                    .flatMap(t -> {
```

Comment:
reduce nesting by moving the inner flatmap below the outer flatmap

Reply:
that cannot be possible because the inner flatMap depends on uuid of the outer flatMap.